test_html_multi_line_copyright fails when SOURCE_DATE_EPOCH is set for 2024

### Describe the bug

While updating sphinx to 8.1.3 in Fedora, we noticed the test `test_html_multi_line_copyright` started to fail at the beginning of the new year 2025. Upon further inspection we find out that SOURCE_DATE_EPOCH in Fedora is based on the last commit date (in our case November 2024). This causes problems because [this condition](https://github.com/sphinx-doc/sphinx/commit/a84f7009acd4c51642e10a2cd15cea415860c0a7#diff-5d140f836bc002529fa8bd76ae49b359f0546739c80e44a4c396fa5eb8d8e2d8R636) is not met and sphinx then tries to replace `current_year` (2025) with `replace_yr` (2024 - taken from source epoch date) which leads to a test failure, because the test [expects 2025](https://github.com/sphinx-doc/sphinx/blob/master/tests/test_builders/test_build_html_copyright.py#L38).

This can be workarounded on our side, by bumping the commit date, but if somebody updates the hardcoded year in tests, it can hit us again in the future. 

cc @jayaddison @a

### How to Reproduce

- clone the repository
- add `SOURCE_DATE_EPOCH = 1731456000` (that's 13 Nov 2024) to [testenv], setenv part of tox.ini
- run `tox -e py313 -- tests/test_builders/test_build_html_copyright.py -vvv`
- see result:

````
    @pytest.mark.sphinx('html', testroot='copyright-multiline')
    def test_html_multi_line_copyright(app):
        app.build(force_all=True)
    
        content = (app.outdir / 'index.html').read_text(encoding='utf-8')
    
        # check the copyright footer line by line (empty lines ignored)
        assert '  &#169; Copyright 2006.<br/>\n' in content
        assert '  &#169; Copyright 2006-2009, Alice.<br/>\n' in content
        assert '  &#169; Copyright 2010-2013, Bob.<br/>\n' in content
        assert '  &#169; Copyright 2014-2017, Charlie.<br/>\n' in content
        assert '  &#169; Copyright 2018-2021, David.<br/>\n' in content
>       assert '  &#169; Copyright 2022-2025, Eve.' in content
E       assert '  &#169; Copyright 2022-2025, Eve.' in '<!DOCTYPE html>\n\n<html lang="en" data-content_root="./">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />\n\n    <title>&lt;no title&gt; &#8212; Project name not set  documentation</title>\n    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=8e8a900e" />\n    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=5c69cfe2" />\n    <script src="_static/documentation_options.js?v=5929fcd5"></script>\n    <script src="_static/doctools.js?v=9bcbadda"></script>\n    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>\n    <link rel="index" title="Index" href="genindex.html" />\n    <link rel="search" title="Search" href="search.html" /> \n  </head><body>\n    <div class="related" role="navigation" aria-label="Related">\n      <h3>Navigation</h3>\n      <ul>\n        <li class="right" style="margin-right: 10px">\n          <a href="genindex.html" title="General Index"\n             accesskey="I">index</a></li>\n        <li class="nav-item nav-item-0"><a href="#">Project name not set  documentation</a> &#187;</li>\n        <li class="nav-item nav-item-this"><a href="">&lt;no title&gt;</a></li> \n      </ul>\n    </div>  \n\n    <div class="document">\n      <div class="documentwrapper">\n        <div class="bodywrapper">\n          <div class="body" role="main">\n            \n  \n\n            <div class="clearer"></div>\n          </div>\n        </div>\n      </div>\n      <div class="sphinxsidebar" role="navigation" aria-label="Main">\n        <div class="sphinxsidebarwrapper">\n  <div role="note" aria-label="source link">\n    <h3>This Page</h3>\n    <ul class="this-page-menu">\n      <li><a href="_sources/index.rst.txt"\n            rel="nofollow">Show Source</a></li>\n    </ul>\n   </div>\n<search id="searchbox" style="display: none
" role="search">\n  <h3 id="searchlabel">Quick search</h3>\n    <div class="searchformwrapper">\n    <form class="search" action="search.html" method="get">\n      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>\n      <input type="submit" value="Go" />\n    </form>\n    </div>\n</search>\n<script>document.getElementById(\'searchbox\').style.display = "block"</script>\n        </div>\n      </div>\n      <div class="clearer"></div>\n    </div>\n    <div class="related" role="navigation" aria-label="Related">\n      <h3>Navigation</h3>\n      <ul>\n        <li class="right" style="margin-right: 10px">\n          <a href="genindex.html" title="General Index"\n             >index</a></li>\n        <li class="nav-item nav-item-0"><a href="#">Project name not set  documentation</a> &#187;</li>\n        <li class="nav-item nav-item-this"><a href="">&lt;no title&gt;</a></li> \n      </ul>\n    </div>\n    <div class="footer" role="contentinfo">\n    \n      &#169; Copyright 2006.<br/>\n    \n      &#169; Copyright 2006-2009, Alice.<br/>\n    \n      &#169; Copyright 2010-2013, Bob.<br/>\n    \n      &#169; Copyright 2014-2017, Charlie.<br/>\n    \n      &#169; Copyright 2018-2021, David.<br/>\n    \n      &#169; Copyright 2022-2024, Eve.\n    \n      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 8.2.0+/74ec22047.\n    </div>\n  </body>\n</html>'
````

### Environment Information

```text
N/A
```


### Sphinx extensions

_No response_

### Additional context

_No response_

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -58,3 +58,6 @@ Bugs fixed
 
 Testing
 -------
+
+* #13224: Correctness fixup for ``test_html_multi_line_copyright``.
+  Patch by Colin Watson, applied by James Addison.

--- a/tests/test_builders/test_build_html_copyright.py
+++ b/tests/test_builders/test_build_html_copyright.py
@@ -9,6 +9,13 @@ LT_NEW = (2009, *LT[1:], LT.tm_zone, LT.tm_gmtoff)
 LOCALTIME_2009 = type(LT)(LT_NEW)
 
 
+@pytest.fixture
+def no_source_date_year(monkeypatch):
+    with monkeypatch.context() as m:
+        m.delenv('SOURCE_DATE_EPOCH', raising=False)
+        yield
+
+
 @pytest.fixture(
     params=[
         1199145600,  # 2008-01-01 00:00:00
@@ -24,7 +31,7 @@ def source_date_year(request, monkeypatch):
 
 
 @pytest.mark.sphinx('html', testroot='copyright-multiline')
-def test_html_multi_line_copyright(app):
+def test_html_multi_line_copyright(no_source_date_year, app):
     app.build(force_all=True)
 
     content = (app.outdir / 'index.html').read_text(encoding='utf-8')

--- a/tests/test_builders/test_build_html_copyright.py
+++ b/tests/test_builders/test_build_html_copyright.py
@@ -11,9 +11,11 @@ LOCALTIME_2009 = type(LT)(LT_NEW)
 
 @pytest.fixture
 def no_source_date_year(monkeypatch):
-    """Explicitly clear SOURCE_DATE_EPOCH from the environment; this
+    """
+    Explicitly clear SOURCE_DATE_EPOCH from the environment; this
     fixture can be used to ensure that copyright substitution logic
-    does not occur during selected test cases."""
+    does not occur during selected test cases.
+    """
     with monkeypatch.context() as m:
         m.delenv('SOURCE_DATE_EPOCH', raising=False)
         yield

--- a/tests/test_builders/test_build_html_copyright.py
+++ b/tests/test_builders/test_build_html_copyright.py
@@ -11,6 +11,9 @@ LOCALTIME_2009 = type(LT)(LT_NEW)
 
 @pytest.fixture
 def no_source_date_year(monkeypatch):
+    """Explicitly clear SOURCE_DATE_EPOCH from the environment; this
+    fixture can be used to ensure that copyright substitution logic
+    does not occur during selected test cases."""
     with monkeypatch.context() as m:
         m.delenv('SOURCE_DATE_EPOCH', raising=False)
         yield

